src/javatests/com/google/cloud/bigquery/dwhassessment/extractiontool/executor/SaveCheckerImplTest.java

```diff
@@ -45,6 +46,24 @@ public void setUp() throws IOException {
         new SaveCheckerImpl(ImmutableMap.of(SCRIPT_NAME, ImmutableList.of("testTimestampColumn")));
   }
 
+  @Test
+  public void findFinishedRecords_onlyReturnsCorrectOnes() throws IOException {
```

Comment:
Would be nice to add more tests covering edge cases:
* Empty input
* No matching files
* All files matches

Reply:
Added tests for the first; the latter two cases aren't qualitatively different from the canonical workflow.